Website/src/rendering/src/components/SessionInformation.tsx

```diff
@@ -80,6 +131,41 @@ const SessionInformation = (props: SessionInformationProps): JSX.Element => {
               />
             </div>
 
+            {props.fields.Rooms && props.fields.Rooms.length > 0 && (
```

Comment:
In order to keep the JSX as readable as possible, it is a good practice to avoid adding logic in it. Same concept as cshtml and ascx files in the .net world. A suggestion is to extract the bits that are rendered given conditions to `const`s like this for all pieces that should be conditionally rendered:

```
const room = props.fields.Rooms && props.fields.Rooms.length > 0
  ? (
    <div className="bg-gray-light p-2 flex align-middle content-center">
      <span>
        <FontAwesomeIcon className="icon h-4 mr-2 inline text-black" icon={faDoorOpen} />
      </span>
      <Text
        className="align-middle content-center"
        tag="p"
        field={props.fields.Rooms[0].fields.Name}
      ></Text>
    </div>
  )
  : undefined;

return (
  <room />
);
```

Reply:
I see this issue was not resolved. Are there plans to avoid having our JSX become harder to read and understand by implementing the above suggestion?

Reply:
created new task
https://dev.azure.com/sitecoredst/Demo/_workitems/edit/6537